### PR TITLE
Add Unit Tests for the New Order transaction.

### PR DESCRIPTION
**Pull request by psudheer21**
Summary:
    This unit test ensures that the NewOrder transaction populates the
    correct tables properly. It does this by initializing the values of the
    required tables to a fixed known value, executing the transaction with
    known arguments and ensuring that the tables are populated with the
    required values.

### CODE REVIEW

File: src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java [hunk 1]
```diff
@@ -303,11 +304,17 @@ private void newOrderTransaction(int w_id, int d_id, int c_id,
 
         stockQuantities[ol_number - 1] = s_quantity;
 
+        if (ol_i_id == 1) {
```
Comment (m-iancu) [inline src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java]: Do we need this log here or was it just for debugging?

File: src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java [hunk 2]
```diff
@@ -303,11 +304,17 @@ private void newOrderTransaction(int w_id, int d_id, int c_id,
...
+          LOG.info("Old s_quant is " + s_quantity + " val " + ol_quantity);
+        }
         if (s_quantity - ol_quantity >= 10) {
           s_quantity -= ol_quantity;
         } else {
           s_quantity += -ol_quantity + 91;
         }
+        if (ol_i_id == 1) {
```
Comment (m-iancu) [inline src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java]: ditto.

File: src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java [hunk 3]
```diff
@@ -391,4 +398,254 @@ private void newOrderTransaction(int w_id, int d_id, int c_id,
...
+  // remote_cnt values.
+  void AssertStockValues(Connection conn, int count,
+                         int[] itemIDs, int[] supplierWhIds,
+                         int[] qtyArr, int[] ytdArr,
+                         int[] orderCntArr, int[] remoteCntArr) throws Exception {
+    try {
+      Statement stmt = conn.createStatement();
+      for (int i = 0; i < count; ++i) {
+        int itemId = itemIDs[i];
+        int supplieWhId = supplierWhIds[i];
```
Comment (m-iancu) [inline src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java]: nit: `supplierWhId`?

**Review by m-iancu — COMMENTED** [review]
Left some minor comments.

File: src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java (hunk 1 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java]: Removed

File: src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java (hunk 2 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java]: Done.

File: src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java (hunk 3 above)
Comment (psudheer21) [reply src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java]: Done

**Review by m-iancu — APPROVED** [review]
LGTM!

**Pull request merged**